src/general/PlayerInspectInfo.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   A system that manages detecting what the player is pointing with the cursor.
+/// </summary>
+public class PlayerInspectInfo : Node
+{
+    /// <summary>
+    ///   The distance for detection.
+    /// </summary>
+    [Export]
+    public float RayastDistance = 1000;
+
+    private readonly List<RaycastResult> hits = new();
+    private readonly HashSet<RaycastResult> previousHits = new();
+
+    /// <summary>
+    ///   All inspectable entities the player is pointing.
+    /// </summary>
+    public IReadOnlyList<IInspectableEntity> InspectableEntities =>
+        hits.Select(h => h.Collider).OfType<IInspectableEntity>().ToList();
+
+    public virtual void Process(float delta)
+    {
+        var space = GetViewport().World.DirectSpaceState;
+        var mousePos = GetViewport().GetMousePosition();
+        var camera = GetViewport().GetCamera();
+
+        var from = camera.ProjectRayOrigin(mousePos);
+        var to = from + camera.ProjectRayNormal(mousePos) * RayastDistance;
+
+        hits.Clear();
+
+        space.IntersectRay(hits, from, to);
+
+        foreach (var hit in previousHits.ToHashSet())
```

Comment:
Using the remove all filtering on the previousHits is better I think rather than having to copy the data here.